element_facemap/facial_behavior_estimation.py

```diff
@@ -259,111 +275,110 @@ class FacialSignal(dj.Imported):
-                    dict(
-                        key,
-                        roi_no=roi_no,
-                        singular_value=dataset['motSv'][i],
-                        motmask=dataset['motMask_reshape'][roi_no+1][i],
-                        projection=dataset['motMask_reshape'][roi_no+1][:,:,i],
-                    ) for i in range(n_components)
+        self.Region.insert(
+            [
+                dict(
+                    key,
+                    roi_no=i,
+                    xrange=dataset["rois"][i]["xrange"],
+                    yrange=dataset["rois"][i]["yrange"],
+                    xrange_bin=dataset["rois"][i]["xrange_bin"],
+                    yrange_bin=dataset["rois"][i]["yrange_bin"],
+                    motion=dataset["motion"][i],
                 )
+                for i in range(len(dataset["rois"]))
+            ]
+        )
 
-        # MovieSVD
-        #do_mov_svd = (FacemapTask & key).fetch1('do_mov_svd')
-        #if do_mov_svd:
-        if 'movSv' in params.keys():
-            n_components = self.get_ncomponents(dataset['movSv'])
-            for roi_no in range(len(dataset['rois'])):
-                self.MovieSVD.insert(
-                    dict(
-                        key,
-                        roi_no=roi_no,
-                        singular_value=dataset['movSv'][i],
-                        motmask=dataset['movMask_reshape'][roi_no+1][i],
-                        projection=dataset['movMask_reshape'][roi_no+1][:,:,i],
-                    ) for i in range(n_components)
-                )
+        # MotionSVD
+        if np.any([np.any(x) for x in dataset.get("motSVD", [False])]):
+            entry = []
+            for roi_no in range(len(dataset["rois"])):
+                for i in range(1, dataset["motSVD"][roi_no + 1].shape[1]):
+                    entry.append(
+                        dict(
+                            key,
+                            roi_no=roi_no,
+                            pc_no=i,
+                            singular_value=dataset["motSv"][i] if "motSv" in dataset else None,
+                            motmask=dataset["motMask_reshape"][roi_no + 1][:, :, i],
+                            projection=dataset["motSVD"][roi_no + 1][i],
+                        )
+                    )
+            self.MotionSVD.insert(entry)
 
+        # MovieSVD
+        if np.any([np.any(x) for x in dataset.get("movSVD", [False])]):
+            entry = []
+            for roi_no in range(len(dataset["rois"])):
+                for i in range(1, dataset["movSVD"][roi_no + 1].shape[1]):
+                    entry.append(
+                        dict(
+                            key,
+                            roi_no=roi_no,
+                            pc_no=i,
+                            singular_value=dataset["movSv"][i] if "movSV" in dataset else None,
+                            movmask=dataset["movMask_reshape"][roi_no + 1][:, :, i],
```

Comment:
why is the first ROI skipped?

Reply:
ROI is not skipped. The first element of the MovSVD and MotSVD is reserved for full image’s SVD. This code is currently not supporting the full image SVDs. I’ll get back to that later.